### PR TITLE
Implement utils for working with Sha

### PR DESCRIPTION
**Pull request by GnomedDev**
Split off from #655

### CODE REVIEW

**Comment by riidefi** [comment]
I think this is fine, but the annoying thing with C++ is that a std::array reference isn't like a Rust arrayref, so this code sort of requires that everyone is locked into the Sha1 struct. Perhaps we could provide two aliases? i.e.
```cpp
using Sha1 = std::array<u8, 0x14>;
using Sha1Ref = std::span<u8, 0x14>;
```

**Comment by GnomedDev** [comment]
Being locked into the `Sha1` struct is mostly the point, for type safety.

**Comment by riidefi** [comment]
Okay--while I personally think this is quite brittle and odd for a function on a read-only span of 20 bytes requiring the programmer to memcpy them into a specific structure (when reading them a file is a use case), I can agree that the security footguns of the current approach (with a raw pointer) are worse. I've approved this change.

**Review by riidefi — APPROVED** [review]

File: payload/sp/ShaUtil.hh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <Common.hh>
+
+#include <expected>
+#include <string_view>
```
Comment (vabold) [inline payload/sp/ShaUtil.hh:6]:
```suggestion
#include <expected>
#include <string_view>

#include <Common.hh>
```

File: payload/sp/ShaUtil.cc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#include "ShaUtil.hh"
+#include <vendor/libhydrogen/hydrogen.h>
```
Comment (vabold) [inline payload/sp/ShaUtil.cc:2]: While you're fixing includes:
```suggestion
#include "ShaUtil.hh"

#include <vendor/libhydrogen/hydrogen.h>
```

**Review by vabold — CHANGES_REQUESTED** [review]

File: payload/sp/ShaUtil.hh (hunk 1 above)
Comment (GnomedDev) [reply payload/sp/ShaUtil.hh:6]: Huh, that's weird, I could swear it was common before stdlib, but I've gone back and checked and nope. There are probably much more examples in the project of incorrect formatting that a tidy check should really be written for it.

**Comment by GnomedDev** [comment]
Solved conflicts and header ordering issues.

**Review by vabold — APPROVED** [review]

**Pull request merged**